pkg/vtpm/src/server.cpp

```diff
@@ -499,14 +519,21 @@ serveClient(int sock, sockaddr_in clientAddr)
         cerr << "recv() from " << inet_ntoa(clientAddr.sin_addr)
              << " failed with " << rc << std::endl;
     } else if (recvBytes == CODED_STRM_HDR_LEN) {
-        if (handleRequest(sock, readHdr(hdrBuf)) != 0) {
-            cerr << "Failure processing the request" << std::endl;
+        google::protobuf::uint32 payloadSize = readHdr(hdrBuf);
+        if (payloadSize > 0 && payloadSize <= (MAX_PAYLOAD_SIZE - CODED_STRM_HDR_LEN)) {
+            if (handleRequest(sock, payloadSize) != 0) {
+                cerr << "Failure processing the request" << std::endl;
+            }
+        }
+        else {
+            cerr << "Failure processing the request, too large (maybe adjust MAX_PAYLOAD_SIZE)" << std::endl;
         }
     } else {
        cerr << "recv() received fewer than expected(" << recvBytes
             << ") bytes from: " << inet_ntoa(clientAddr.sin_addr)
             << std::endl;
     }
+    _g_numConn--;
```

Comment:
Dint we need to make the increments and decrements be atomic?

Reply:
I wrote it atomic at first, but then I couldn't think of any case that _g_numConn inc/dec being non-atomic causing a problem, it is not used as index to array or queue or in compare and exchange operation or else. If there is problematic case, I will change it to atomic.

Reply:
You spawn a thread for each connection, and each thread will decrement (by doing a read then write) and those can happen at the same time y different threads AFAICT.